plugin.js

```diff
@@ -5,7 +5,7 @@ const urijs = require('uri-js')
 
 function plugin (fastify, options, next) {
   fastify.decorateRequest('urlData', function (key) {
-    const scheme = (this.headers[':scheme'] ? this.headers[':scheme'] + ':' : '') + '//'
+    const scheme = (this.headers[':scheme'] ? this.headers[':scheme'] + ':' : this.protocol + ':') + '//'
```

Comment:
The scheme is the text preceding the colon (`:`) according to the RFC https://datatracker.ietf.org/doc/html/rfc1738#section-2.1

> In general, URLs are written as follows:
>
>       <scheme>:<scheme-specific-part>
>
>   A URL contains the name of the scheme being used (<scheme>) followed
>   by a colon and then a string (the <scheme-specific-part>) whose
>   interpretation depends on the scheme.
>
>   Scheme names consist of a sequence of characters. The lower case
>   letters "a"--"z", digits, and the characters plus ("+"), period
>   ("."), and hyphen ("-") are allowed. For resiliency, programs
>   interpreting URLs should treat upper case letters as equivalent to
>   lower case in scheme names (e.g., allow "HTTP" as well as "http").

Thus, in `http://example.com` the scheme is `http` not `http://`.

Reply:
it will return the same scheme as before - it returns result of urijs.parse().scheme. The question is about this variable name